src/ens-avatar/src/specs/erc721.ts

```diff
@@ -42,10 +43,15 @@ export default class ERC721 {
       }
       return JSON.parse(_resolvedUri);
     }
-    const { image_url } = await apiGetUniqueTokenImage(
-      contractAddress,
-      tokenID
-    );
-    return { image: image_url };
+
+    let image;
+    try {
+      const data = await apiGetUniqueTokenImage(contractAddress, tokenID);
+      image = data.image_url;
+    } catch (error) {
+      const data = await getNFTByTokenId({ contractAddress, tokenId: tokenID });
+      image = data.previews?.image_medium_url;
```

Comment:
`getNFTByTokenId` swallows any errors it gets and does not return a value for `data` if so - can we add optional chaining on `data` also: `data?.previews?.image_medium_url`